PCES EOF Exception

https://swirldslabs.slack.com/archives/C03EPEDDJ9Y/p1691563080770269

```
java.io.UncheckedIOException: error encountered while reading from the PCES
	at com.swirlds.platform.event.preconsensus.PreconsensusEventReplayPipeline.replayEvents(PreconsensusEventReplayPipeline.java:151) ~[swirlds-platform-core-0.41.0-SNAPSHOT.jar:?]
	at com.swirlds.platform.event.preconsensus.PreconsensusEventReplayWorkflow.replayPreconsensusEvents(PreconsensusEventReplayWorkflow.java:103) ~[swirlds-platform-core-0.41.0-SNAPSHOT.jar:?]
	at com.swirlds.platform.SwirldsPlatform.replayPreconsensusEvents(SwirldsPlatform.java:1064) ~[swirlds-platform-core-0.41.0-SNAPSHOT.jar:?]
	at com.swirlds.platform.SwirldsPlatform.start(SwirldsPlatform.java:1047) ~[swirlds-platform-core-0.41.0-SNAPSHOT.jar:?]
	at java.util.ArrayList.forEach(ArrayList.java:1511) ~[?:?]
	at java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1092) ~[?:?]
	at com.swirlds.platform.Browser.startPlatforms(Browser.java:953) ~[swirlds-platform-core-0.41.0-SNAPSHOT.jar:?]
	at com.swirlds.platform.Browser.<init>(Browser.java:313) ~[swirlds-platform-core-0.41.0-SNAPSHOT.jar:?]
	at com.swirlds.platform.Browser.launch(Browser.java:541) ~[swirlds-platform-core-0.41.0-SNAPSHOT.jar:?]
	at com.swirlds.platform.Browser.parseCommandLineArgsAndLaunch(Browser.java:490) ~[swirlds-platform-core-0.41.0-SNAPSHOT.jar:?]
	at com.swirlds.platform.Browser.main(Browser.java:972) ~[swirlds-platform-core-0.41.0-SNAPSHOT.jar:?]
Caused by: java.io.EOFException
	at java.io.DataInputStream.readInt(DataInputStream.java:398) ~[?:?]
	at com.swirlds.common.io.streams.AugmentedDataInputStream.readInt(AugmentedDataInputStream.java:169) ~[swirlds-common-0.41.0-SNAPSHOT.jar:?]
	at com.swirlds.platform.event.preconsensus.PreconsensusEventFileIterator.<init>(PreconsensusEventFileIterator.java:59) ~[swirlds-platform-core-0.41.0-SNAPSHOT.jar:?]
	at com.swirlds.platform.event.preconsensus.PreconsensusEventMultiFileIterator.findNext(PreconsensusEventMultiFileIterator.java:68) ~[swirlds-platform-core-0.41.0-SNAPSHOT.jar:?]
	at com.swirlds.platform.event.preconsensus.PreconsensusEventMultiFileIterator.hasNext(PreconsensusEventMultiFileIterator.java:80) ~[swirlds-platform-core-0.41.0-SNAPSHOT.jar:?]
	at com.swirlds.platform.event.preconsensus.PreconsensusEventReplayPipeline.replayEvents(PreconsensusEventReplayPipeline.java:137) ~[swirlds-platform-core-0.41.0-SNAPSHOT.jar:?]
	... 10 more
```

--- a/platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PreconsensusEventFileIterator.java
+++ b/platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PreconsensusEventFileIterator.java
@@ -37,6 +37,7 @@ public class PreconsensusEventFileIterator implements IOIterator<GossipEvent> {
     private boolean hasPartialEvent = false;
     private final CountingStreamExtension counter;
     private GossipEvent next;
+    private boolean streamClosed = false;
 
     /**
      * Create a new iterator that walks over events in a preconsensus event file.
@@ -56,14 +57,24 @@ public class PreconsensusEventFileIterator implements IOIterator<GossipEvent> {
                 new BufferedInputStream(
                         new FileInputStream(fileDescriptor.getPath().toFile())),
                 counter));
-        stream.readInt(); // read the version number, but we don't need it atm
+
+        try {
+            final int fileVersion = stream.readInt();
+            if (fileVersion != PreconsensusEventMutableFile.FILE_VERSION) {
+                throw new IOException("unsupported file version: " + fileVersion);
+            }
+        } catch (final EOFException e) {
+            // Empty file. Possible if the node crashed right after it created this file.
+            stream.close();
+            streamClosed = true;
+        }
     }
 
     /**
      * Find the next event that should be returned.
      */
     private void findNext() throws IOException {
-        while (next == null) {
+        while (next == null && !streamClosed) {
 
             final long initialCount = counter.getCount();
 
@@ -78,7 +89,8 @@ public class PreconsensusEventFileIterator implements IOIterator<GossipEvent> {
                     // This is possible (if not likely) when a node is shut down abruptly.
                     hasPartialEvent = true;
                 }
-                break;
+                stream.close();
+                streamClosed = true;
             }
         }
     }

--- a/platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PreconsensusEventMutableFile.java
+++ b/platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PreconsensusEventMutableFile.java
@@ -33,7 +33,7 @@ import java.nio.file.StandardCopyOption;
  */
 public class PreconsensusEventMutableFile {
     /** the file version to write at the beginning of the file. atm, this is just a placeholder for future changes */
-    private static final int FILE_VERSION = 1;
+    public static final int FILE_VERSION = 1;
 
     /**
      * Describes the file that is being written to.

--- a/platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/event/preconsensus/PreconsensusEventReadWriteTests.java
+++ b/platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/event/preconsensus/PreconsensusEventReadWriteTests.java
@@ -40,11 +40,13 @@ import com.swirlds.platform.test.event.source.StandardEventSource;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.time.Instant;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
+import java.util.NoSuchElementException;
 import java.util.Random;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeAll;
@@ -509,5 +511,21 @@ class PreconsensusEventReadWriteTests {
         for (int i = 0; i < events.size(); i++) {
             assertEventsAreEqual(events.get(i), deserializedEvents.get(i));
         }
+    }
+
+    @Test
+    @DisplayName("Empty File Test")
+    void emptyFileTest() throws IOException {
+        final PreconsensusEventFile file = PreconsensusEventFile.of(0, 0, 100, Instant.now(), testDirectory, false);
+
+        final Path path = file.getPath();
+
+        Files.createDirectories(path.getParent());
+        assertTrue(path.toFile().createNewFile());
+        assertTrue(Files.exists(path));
+
+        final PreconsensusEventFileIterator iterator = file.iterator(Long.MIN_VALUE);
+        assertFalse(iterator.hasNext());
+        assertThrows(NoSuchElementException.class, iterator::next);
     }
 }

